app/gui/src/project-view/components/widgets/FileBrowserWidget/FileBrowserNameBar.vue

```diff
@@ -1,18 +1,159 @@
 <script setup lang="ts">
+import SelectionSubmenu, {
+  SubmenuComponent,
+} from '@/components/GraphEditor/widgets/WidgetSelection/SelectionSubmenu.vue'
+import { SubmenuEntry } from '@/components/GraphEditor/widgets/WidgetSelection/submenuEntry'
 import SvgButton from '@/components/SvgButton.vue'
+import SvgIcon from '@/components/SvgIcon.vue'
+import AutoSizedInput from '@/components/widgets/AutoSizedInput.vue'
+import { injectInteractionHandler, Interaction } from '@/providers/interactionHandler'
+import {
+  FileType,
+  isExtensions,
+  isFileTypes,
+  isGlobAll,
+} from '@/providers/widgetRegistry/configuration'
+import { endOnClick, targetIsOutside } from '@/util/autoBlur'
+import { Opt } from '@/util/data/opt'
+import { computed, ref, useTemplateRef } from 'vue'
+import { Filter } from './fileExtensionFilter'
 
-const filenameInputContents = defineModel<string>({ required: true })
+const filenameInput = defineModel<string>('filenameInput', { required: true })
+const extensionInput = defineModel<string>('extensionInput', {
+  required: true,
+})
 
+const props = defineProps<{
+  writeMode: boolean
+  fileExtensionFilter: Filter
+  displayedExtension: string
+  fileTypes: FileType[]
+  root: Opt<HTMLElement>
+}>()
 const emit = defineEmits<{
   accept: []
+  setFilter: [Filter]
 }>()
+
+// === File Extension Filter ===
+
```

Comment:
It's ok to go, but I would still at some point make it a component separate from FileBrowserNameBar - just not not have everything prefixed with `fileExtension` ;)

Reply:
Ugh, I tried doing that, but splitting the components is difficult because they are interconnected. I will leave it as it is.